ui/src/api/events/query/get.ts

```diff
@@ -1,168 +1,94 @@
-
-    // Without the guard, TS thinks filterItem may be undefined
-    if (!filterItem) {
-      return console.error("filterItem is not defined");
-    }
-
-    let queryField: string;
-    let queryValue: string;
-
-    if (field === "AFTER" || field === "BEFORE") {
-      const date = filters[field]?.value;
-      if (!date) {
-        throw new Error("date is not defined in date filter");
-      }
-      queryField = "CREATED";
-      queryValue = date.toISOString();
-    } else {
-      const value = filters[field]?.value;
-      if (!value) {
-        throw new Error("filter value is not defined");
-      }
-      queryField = field;
-      queryValue = value;
-    }
-
-    query = query.concat(
-      `&filter.field=${queryField}&filter.type=${filterItem.type}&filter.val=${queryValue}`
-    );
-  });
-
-  return query;
-};
-
-// TODO: this is also duplicated, abstract it (url needs to be an argument)
 const getUrl = ({
   namespace,
   baseUrl,
-  limit,
-  offset,
   filters,
 }: {
-  baseUrl?: string;
   namespace: string;
-  limit: number;
-  offset: number;
-  filters?: FiltersObj;
+  baseUrl?: string;
+  filters?: FiltersSchemaType;
 }) => {
-  let url = `${
-    baseUrl ?? ""
-  }/api/namespaces/${namespace}/events?limit=${limit}&offset=${offset}`;
+  let url = `${baseUrl ?? ""}/api/v2/namespaces/${namespace}/events/history`;
+
   if (filters) {
-    url = url.concat(getFilterQuery(filters));
+    url = url.concat(buildSearchParamsString(filters));
   }
+
   return url;
 };
 
 export const getEvents = apiFactory({
   url: getUrl,
   method: "GET",
-  schema: EventsListSchema,
+  schema: EventsListResponseSchema,
 });
 
 const fetchEvents = async ({
-  queryKey: [{ apiKey, namespace, limit, offset, filters }],
+  queryKey: [{ apiKey, namespace, filters }],
 }: QueryFunctionContext<ReturnType<(typeof eventKeys)["eventsList"]>>) =>
   getEvents({
     apiKey,
-    urlParams: { namespace, limit, offset, filters },
+    urlParams: { namespace, filters },
   });
 
-export const useEvents = ({
-  limit,
-  offset,
-  filters,
-}: {
-  limit: number;
-  offset: number;
-  filters: FiltersObj;
-}) => {
+export const useEvents = ({ enabled = true, filters }: EventsListParams) => {
   const apiKey = useApiKey();
   const namespace = useNamespace();
 
   if (!namespace) {
     throw new Error("namespace is undefined");
   }
 
-  return useQueryWithPermissions({
+  /**
+   * The API returns data as an infinite list, which means it returns a cursor in form of a timestamp
+   * to the previous page of data. The end of the list is not known until the last page is reached and
+   * the cursor is null.
+   *
+   * The API only returns navigation into one direction, which means we always have to start with querying
+   * the most recent logs and then navigate to older ones. It is not possible to start at a specific time
+   * and then move to more recent logs.
+   */
+  const queryReturn = useInfiniteQueryWithPermissions({
```

Comment:
Good point. I switched to the simple query hook in 9c9ecde91. 

Not only does this make the name less confusing, it also makes updating the cache much simpler because we don't have to worry about pages.